kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java

```diff
@@ -150,6 +150,11 @@ public void populateKamelet(
         setSpecDependencies(kamelet.getSpec(), steps, metadata);
         setSpecDefinition(kamelet, parameters);
         kamelet.getSpec().getTemplate().setBeans((List<Bean>) metadata.getOrDefault("beans", null));
+        final String metadataDescription = metadata.getOrDefault("description","").toString();
+        final String specDesription = kamelet.getSpec().getDefinition().getDescription();
+        if (!metadataDescription.equals("") || specDesription == null) {
```

Comment:
a good habit is to put the constant first to avoid potential NPE `"".equals(metadataDescription)`